src/lib/cimbar_js/cimbar_recv_js.cpp

```diff
@@ -0,0 +1,219 @@
+/* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
+#include "cimbar_recv_js.h"
+
+#include "cimb_translator/Config.h"
+#include "encoder/Decoder.h"
+#include "encoder/escrow_buffer_writer.h"
+#include "extractor/Extractor.h"
+#include "fountain/fountain_decoder_sink.h"
+#include "serialize/str_join.h"
+#include "util/Timer.h"
+
+#include <opencv2/opencv.hpp>
+
+#include <algorithm>
+#include <memory>
+
+
+namespace {
+
+	std::shared_ptr<fountain_decoder_sink> _sink;
+	std::string _reporting;
+
+	cv::Mat _debugFrame;
+
+	TimeAccumulator _tScanExtract;
+	TimeAccumulator _tImgDecode;
+
+	// settings
+	unsigned _colorBits = 2;
+	int _modeVal = 68;
+
+	bool legacy_mode()
+	{
+		return _modeVal == 4;
```

Comment:
Hoping to clean up this "legacy_mode" ugliness when I refactor the config system...